### PR TITLE
TEST-#6708: Create test files using `tmp_path` fixture

### LINKED ISSUE #6708: Create test files using `tmp_path` fixture

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

This change will ensure that test files no longer clutter up the current working directory.

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6708 
- [x] tests ~added and~ passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by anmyachev** [comment]
@YarShev @dchigarev ready for review

File: modin/conftest.py [hunk 1]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
-        "test_read_csv_regular",
-        "test_read_csv_blank_lines",
-        "test_read_csv_yes_no",
-        "test_read_csv_nans",
-        "test_read_csv_bad_lines",
-    ]
+def TestReadCSVFixture(tmp_path_factory):
+    tmp_path = tmp_path_factory.mktemp("TestReadCSVFixture")
+
+    creater = _make_csv_file(data_dir=tmp_path)
```
Comment (YarShev) [inline modin/conftest.py:300]:
```suggestion
    creator = _make_csv_file(data_dir=tmp_path)
```

File: modin/conftest.py [hunk 2]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
     # each xdist worker spawned in separate process with separate namespace and dataset
-    pytest.csvs_names = {file_id: get_unique_filename() for file_id in files_ids}
+    pytest.csvs_names = {}
     # test_read_csv_col_handling, test_read_csv_parsing
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_regular"],
-    )
+    pytest.csvs_names["test_read_csv_regular"] = creater()
```
Comment (anmyachev) [inline modin/conftest.py:304]:
```suggestion
    pytest.csvs_names["test_read_csv_regular"] = creator()
```

File: modin/conftest.py [hunk 3]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
     # test_read_csv_parsing
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_yes_no"],
+    pytest.csvs_names["test_read_csv_yes_no"] = creater(
```
Comment (anmyachev) [inline modin/conftest.py:306]:
```suggestion
    pytest.csvs_names["test_read_csv_yes_no"] = creator(
```

File: modin/conftest.py [hunk 4]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
         additional_col_values=["Yes", "true", "No", "false"],
     )
     # test_read_csv_col_handling
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_blank_lines"],
+    pytest.csvs_names["test_read_csv_blank_lines"] = creater(
```
Comment (anmyachev) [inline modin/conftest.py:310]:
```suggestion
    pytest.csvs_names["test_read_csv_blank_lines"] = creator(
```

File: modin/conftest.py [hunk 5]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_nans"],
+    pytest.csvs_names["test_read_csv_nans"] = creater(
         add_blank_lines=True,
         additional_col_values=["<NA>", "N/A", "NA", "NULL", "custom_nan", "73"],
     )
     # test_read_csv_error_handling
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_bad_lines"],
+    pytest.csvs_names["test_read_csv_bad_lines"] = creater(
```
Comment (anmyachev) [inline modin/conftest.py:319]:
```suggestion
    pytest.csvs_names["test_read_csv_bad_lines"] = creator(
```

File: modin/conftest.py [hunk 6]
```diff
@@ -295,65 +294,44 @@ def pytest_runtest_call(item):
...
     # test_read_csv_col_handling
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_blank_lines"],
+    pytest.csvs_names["test_read_csv_blank_lines"] = creater(
         add_blank_lines=True,
     )
     # test_read_csv_nans_handling
-    _make_csv_file(filenames)(
-        filename=pytest.csvs_names["test_read_csv_nans"],
+    pytest.csvs_names["test_read_csv_nans"] = creater(
```
Comment (anmyachev) [inline modin/conftest.py:314]:
```suggestion
    pytest.csvs_names["test_read_csv_nans"] = creator(
```

**Review by YarShev — APPROVED** [review]

**Pull request merged**